x-pack/plugins/maps/server/kibana_server_services.ts

```diff
@@ -14,3 +17,16 @@ export const setInternalRepository = (
   internalRepository = createInternalRepository();
 };
 export const getInternalRepository = () => internalRepository;
+
+let indexPatternsService: IndexPatternsCommonService;
+export const setIndexPatternsService = async (
+  indexPatternsServiceFactory: IndexPatternsServiceStart['indexPatternsServiceFactory'],
+  elasticsearchClient: ElasticsearchClient
+) => {
+  indexPatternsService = await indexPatternsServiceFactory(
+    // @ts-expect-error
```

Comment:
the `@ts-expect-error` should be addressed. It looks like `createInternalRepository` should be wrapped in a `SavedObjectsClient` - I'm not sure why its not provided this way.  the client is a thin wrapper around the repository hence the reason for the type error yet the code still works.

Reply:
@mattkime Would it be worth creating a small issue to track this or would you prefer to fix the typing issue in this PR?

Reply:
After talking to core - this is only going to get index patterns defined in the default space. You need all index patterns, correct? We may need to tweak the index patterns api.

Reply:
I think we'd prefer getting all index patterns, yeah. Follow-on issue?

Reply:
@mattkime Also, I was able to wrap `getInternalRepository()` per your suggestion